backend/substrapp/views/objective.py

```diff
@@ -231,10 +231,9 @@ def data(self, request, *args, **kwargs):
 
     @action(detail=True, methods=['GET'])
     def leaderboard(self, request, pk):
-        validate_pk(pk)
-
         sort = request.query_params.get('sort', 'desc')
         try:
+            validate_pk(pk)
```

Comment:
`validate_pk(pk)` should be in its own try/except.

A try clause  should only try on one call.
Maybe creating a dedicated method:
```python
def method(self, request, pk):
    validate_pk(pk)
    sort = request.query_params.get('sort', 'desc')
    validate_sort(sort)
```

is better.
Then try/except on this method for handling all the possible cases.

Reply:
Thank you, I will update my PR :)

Reply:
I don't know if we need to create a dedicated method for this.
I'd rather prefer to use a try/except